src/library.js

```diff
@@ -5214,6 +5214,24 @@ LibraryManager.library = {
     return thisProgram || './this.program';
 #endif
   },
+
+  $autoResumeAudioContext: function(ctx) {
+    ['keydown', 'mousedown'].forEach(function(event) {
+      // Browsers will only allow audio to play after a user interaction. Listen
+      // for a mouse click on the document and on a canvas, if one exists.
+      document.addEventListener(event, function resumeAudioOnDocument() {
+        if (ctx.state === 'suspended') ctx.resume();
+        document.removeEventListener(event, resumeAudioOnDocument);
+      });
+      var canvas = document.getElementById('canvas');
```

Comment:
Yikes, this is a bit nasty and can be brittle.

It would perhaps be better if we created an OpenAL "extension function" e.g.
```c++
void emscripten_alAutoResumeAudioContext(const char *target);
```
then users could call that function with the CSS selector of their choice. That would have the benefit that if users already have their custom enable code in place, this code would not be doubly introduced. Also it would avoid possibly misfinding the canvas above.

Reply:
That also sounds worthwhile, good idea!

However I still want to have as good default behavior as possible. Doing document and canvas is enough to get our tests passing, and probably most user projects out there? So I'd like to keep this approach, but we can also add a new API (maybe not necessarily OpenAL specific? anyhow, can discuss separately).